src/main/scala/org/broadinstitute/dsde/workbench/sam/dataAccess/PostgresAccessPolicyDAO.scala

```diff
@@ -951,6 +952,81 @@ class PostgresAccessPolicyDAO(
     }
   }
 
+  // TODO write the correct query
+  // Return value: [(policyToUpdate, policyToRemove)]
+  override def findAffectedPolicyGroups(
+      resourceId: FullyQualifiedResourceId,
+      samRequestContext: SamRequestContext
+  ): IO[List[(FullyQualifiedPolicyId, FullyQualifiedPolicyId)]] =
+    readOnlyTransaction("findAffectedPolicyGroups", samRequestContext) { implicit session =>
+      val g = GroupTable.syntax("g")
+      val pg = GroupTable.syntax("pg")
+      val gm = GroupMemberTable.syntax("gm")
+      val p = PolicyTable.syntax("p")
+      val rt = ResourceTypeTable.syntax("rt")
+      val r = ResourceTable.syntax("r")
```

Comment:
I suggest naming these something more readable and include whether they are representing rows from the container or member policy. Same with fullyQualifiedPolicyId1 vs fullyQualifiedPolicyId2. One is the container and one is the member. Naming them better will help keep things straight.